src/native/eventpipe/ds-ipc-win32.c

```diff
@@ -41,17 +41,20 @@
 #define DS_EXIT_BLOCKING_PAL_SECTION
 #endif
 
+#ifndef __cplusplus
 #undef ep_rt_object_alloc
 #define ep_rt_object_alloc(obj_type) ((obj_type *)calloc(1, sizeof(obj_type)))
```

Comment:
`calloca` zero initializes. `new (nothrow)` does not zero initialize. Is the calloc path paying unnecessary overhead? Should it use malloc instead?

Reply:
We allocate structs through these API's and the behaviour of allocating a new (nothrow) struct using empty constructor it will zero memory, so should behave as calloca.

Reply:
>  allocating a new (nothrow) struct using empty constructor it will zero memory

I do not think `new (nothrow)` zero-initializes anything.

Reply:
If you new a struct and apply empty constructor it will zero-initialize it in C++11, so doing this:

typedef struct _test {
    int a;
    int b;
    int c;
} test;

test * x = new (nothrow)test ()

will zero initialize but just doing:

test * x = new (nothrow)test

will not.

Reply:
Ah ok. I did not know that.